gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -76,15 +76,37 @@ public static boolean isEnabled(State state) {
         .valueOf(state.getProp(ConfigurationKeys.METRICS_ENABLED_KEY, ConfigurationKeys.DEFAULT_METRICS_ENABLED));
   }
 
-  public static GobblinMetrics get(String id) {
+  /**
+   * Get a {@link GobblinMetrics} instance with the given ID.
+   *
+   * @param id the given {@link GobblinMetrics} ID
+   * @return a {@link GobblinMetrics} instance
+   */
+  public synchronized static GobblinMetrics get(String id) {
```

Comment:
This doesn't need to be synchronized, get(String, MetricContext, List<Tag<?>>) already is. Although it might not be important? I don't know what the overhead of synchronizing calls is.